src/library/trackcollection.cpp

```diff
@@ -87,6 +87,13 @@ void TrackCollection::relocateDirectory(QString oldDir, QString newDir) {
     GlobalTrackCacheLocker().relocateCachedTracks(&m_trackDao);
 }
 
+QList<TrackId> TrackCollection::getAndEnsureTrackIds(
```

Comment:
The function getTrackIds() has already a name that doesn't fit.

Since it is a lookup and mapping "resolveTrackIds()" comes into my mind. But I didn't check how to combine it with the 2 options *addMissingTracks* and *unhideHiddenTracks*.

Reply:
The boolean parameter should be replace by an enum that represents a bitmask with the options ADD_MISSING_TRACKS and UNHIDE_HIDDEN_TRACKS.

The function name should contain the word "File", because we are going from files to ids.

But I'm still searching for the right verb that characterizes this operation in a single word. It's difficult, because the operation is so manifold. How about `emplaceTrackFiles()`? @Be-ing Any suggestions from a native speaker?

Reply:
Maybe "resolveTrackIds()" it reads well in combination with the descriptive enum parameter. 
I will try that.